### PR TITLE
Loan refactor

### PR DESCRIPTION
**Pull request by szerintedmi**
- refactor storage. Changes reduced gas costs:
  - first new loan:  306,555 to 197,196
  - consecutive new loan: 291,574 to 167,196
  - first repay: 106,534 to 105,354
  - cons. repay: ?
  - first collect:  73,333 to 71,695
  - cons. collect: 64,683 to 63,620
- new UI helper fx-s:
  - `getLoans(offset)`
  - `getLoanCountForAddress(borrower)` & `getLoansForAddress(borrower, offset)` 
- using div instead of roundedDiv for loan calculations - attempt to simplify UI calculations too as we don't need roundedDiv here (unlike on exchange orderfills)
- allow zero or negative interest on loans (discountRate >= 1)
- split Defaulted loan state into Defaulted (not stored, returned only via getters) and Collected
- finished pending unit tests

### CODE REVIEW

File: contracts/LoanManager.sol [hunk 1]
```diff
@@ -246,8 +249,11 @@ contract LoanManager is Restricted {
             (loanAmount, interestAmount) = calculateLoanValues(product, loan.repaymentAmount);
             uint disbursementTime = loan.maturity - product.term;
 
+            LoanState loanState =
```
Comment (phraktle) [inline contracts/LoanManager.sol]: suggest moving this duplicate logic into a common function.

File: test/helpers/loanTestHelpers.js [hunk 2]
```diff
@@ -374,13 +374,13 @@ async function calcLoanValues(rates, product, collateralWei) {
     ret.repaymentAmount = ret.tokenValue
         .mul(product.collateralRatio)
         .div(ppmDiv)
-        .round(0, BigNumber.ROUND_HALF_UP);
+        .round(0, BigNumber.ROUND_DOWN);
```
Comment (phraktle) [inline test/helpers/loanTestHelpers.js]: i don't see the motivation for this change. if the math works out with `ROUND_HALF_UP` why not just keep that?

File: contracts/LoanManager.sol (hunk 1 above)
Comment (szerintedmi) [reply contracts/LoanManager.sol]: I was wondering to move the whole  result generation into one function (for one loan). But I wasn't sure passing uint[10] arrays on heap is a good idea. Although this is an external view only fx so might be OK.
What you think?

File: test/helpers/loanTestHelpers.js (hunk 2 above)
Comment (szerintedmi) [reply test/helpers/loanTestHelpers.js]: The ultimate test case to demonstrate the requirement is on the UI (see cypress test)
The aim is to have the exact actual loanAmount, repaymentAmount  calculated and shown for the user on the new loan form. 
It was very complex to achieve that with rounded divs on UI (actually I gave it up). And for loans we don't need rounded divs (unlike with fill on exchange). 
Open for change but then the UI calc has to be updated (see `newLoanForm.js` on 'loans_refactor' branch augmint-web )

File: contracts/LoanManager.sol (hunk 1 above)
Comment (szerintedmi) [reply contracts/LoanManager.sol]: OK, just pushed it with a new common function: `getLoanTuple`

**Pull request merged**